### DISCUSSION TITLE
Would it be possible to present an example of how to use DBusCXX with QT UI?

### DISCUSSION THREAD

**Discussion by carlosdelfino** [dbus-cxx/dbus-cxx#79] [General]
I'm learning QT, and I've chosen to use DBusCxx to unify my system's interprocess communication service.

I still have little programming experience with C++, much less with QT and of course none with DBus, I've used a lot of XMLRPC with Java, PHP and even Delphi a decade ago.

On one side is a shared library that is loaded by a third party application, where I didn't have any problems.

On the other side a QT interface that acts as a server for the DBus. But with QT I'm still quite confused, would it be possible to show me an example of how to use DBusCxx to meet 2 or more requests from a DBus system?

As far as I've gone at the moment I've tried for a Singleton, to avoid recreating the `Dispacher`, `Connection`, but I'm pretty confused by the results, and I still haven't had success. Now the big problem is the fact that QT (Qt Creator 4.5.2, Based on Qt 5.9.5 (GCC 7.3.0, 32 bit)) doesn't seem to do very well with the DBusCxx code which is C+17, I'm using the C++1z flag in QMake to parameterize the compilation.
 
This Monday I might choose to use QDBus, but I would like to insist a little more with DBusCxx in QT.

**Comment by rm5248** [comment]
Have you looked at the examples directory?  There are standalone programs there that show you how to handle messages(both method calls and signals).

There are some notes on dispatching here: https://dbus-cxx.github.io/dispatching.html

**Comment by carlosdelfino** [comment]
The difficulty I'm having is not the pure use of DBus, the examples that are offered were great to introduce the subject and I was successful in using them.

The difficulty lies in the lack of more advanced experience, the integration of DBusCxx into a more complex application structure.

But I'm going to do some tests here, as soon as I'm successful I'll share the code here on GitHub and post it on the forum.

Thanks.
